presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkStaticQueryExecution.java

```diff
@@ -0,0 +1,266 @@
+            CollectionAccumulator<PrestoSparkShuffleStats> shuffleStatsCollector,
+            PrestoSparkTaskExecutorFactory taskExecutorFactory,
+            PrestoSparkTaskExecutorFactoryProvider taskExecutorFactoryProvider,
+            QueryStateTimer queryStateTimer,
+            WarningCollector warningCollector,
+            String query,
+            PlanAndMore planAndMore,
+            Optional<String> sparkQueueName,
+            Codec<TaskInfo> taskInfoCodec,
+            JsonCodec<PrestoSparkTaskDescriptor> sparkTaskDescriptorJsonCodec,
+            JsonCodec<PrestoSparkQueryStatusInfo> queryStatusInfoJsonCodec,
+            JsonCodec<PrestoSparkQueryData> queryDataJsonCodec,
+            PrestoSparkRddFactory rddFactory,
+            TransactionManager transactionManager,
+            PagesSerde pagesSerde,
+            PrestoSparkExecutionExceptionFactory executionExceptionFactory,
+            Duration queryTimeout,
+            long queryCompletionDeadline,
+            PrestoSparkMetadataStorage metadataStorage,
+            Optional<String> queryStatusInfoOutputLocation,
+            Optional<String> queryDataOutputLocation,
+            TempStorage tempStorage,
+            NodeMemoryConfig nodeMemoryConfig,
+            Set<PrestoSparkServiceWaitTimeMetrics> waitTimeMetrics,
+            Optional<ErrorClassifier> errorClassifier,
+            PrestoSparkPlanFragmenter planFragmenter,
+            Metadata metadata,
+            PartitioningProviderManager partitioningProviderManager)
+    {
+        super(
+                sparkContext,
+                session,
+                queryMonitor,
+                taskInfoCollector,
+                shuffleStatsCollector,
+                taskExecutorFactory,
+                taskExecutorFactoryProvider,
+                queryStateTimer,
+                warningCollector,
+                query,
+                planAndMore,
+                sparkQueueName,
+                taskInfoCodec,
+                sparkTaskDescriptorJsonCodec,
+                queryStatusInfoJsonCodec,
+                queryDataJsonCodec,
+                rddFactory,
+                transactionManager,
+                pagesSerde,
+                executionExceptionFactory,
+                queryTimeout,
+                queryCompletionDeadline,
+                metadataStorage,
+                queryStatusInfoOutputLocation,
+                queryDataOutputLocation,
+                tempStorage,
+                nodeMemoryConfig,
+                waitTimeMetrics,
+                errorClassifier,
+                planFragmenter,
+                metadata,
+                partitioningProviderManager);
+    }
+
+    @Override
+    protected List<Tuple2<MutablePartitionId, PrestoSparkSerializedPage>> doExecute()
+            throws SparkException, TimeoutException
+    {
+        SubPlan rootFragmentedPlan = getFragmentedPlan();
+        this.finalFragmentedPlan = rootFragmentedPlan;
```

Comment:
maybe have a setter method that checks that this has not been set yet, and throws an exception if it has.

Reply:
oh but if you are doing a get and set kind of operation, then you should use AtomicReference instead of volatile, and then use something like getAndUpdate() to set the value or throw an exception.

Reply:
We are not doing get and set kind of operation. So I dont think its needed